### DISCUSSION TITLE
Help with ShowDialog() factories

### DISCUSSION THREAD

**Discussion by kezara** [canton7/Stylet#329] [Q&A]
I have lot of dialogs in application in different places, and at all that places I call IContainer and IWindowManager. I'd like to avoid calling IContainer and IWindowManager in hundreds of places just to show dialog,  so I was trying to figure out what you were written in [documentation](https://github.com/canton7/Stylet/wiki/StyletIoC-Factories#the-problem), but I couldn't quite understand how to implement factory that will create dialog for different view models.
My idea is to have e.g. some static method that I could call to show me any dialog I want, and to return me a result, something like this
`bool? result = ShowDialogFactory.ShowDialog(ViewModel1);`
`bool? result = ShowDialogFactory.ShowDialog(ViewModel2);`  or similar to that.
Could you please give some hint? Thank you.

**Comment by canton7** [comment]
It's not clear why you need an `IContainer` to show a dialog?

Sure, if you want to have a static class which can show dialogs, just:

```cs
public static class ShowDialogFactory
{
    public static IWindowManager WindowManager { get; set; }
    public static bool? ShowDialog(object viewModel) => WindowManager.ShowDialog(viewModel);
}
```

Then in your bootstrapper's `Configure` method:

```cs
ShowDialogFactory.WindowManager = Container.Get<IWindowManager>();
```

**Reply by kezara** [reply]
Well, I'm using IContainer for instantiating dialog view model when I need it... other solution is to instantiate it manualy... or there is some other solution?

Currently I create dialog like this 

```CSharp
public ViewModel
{
     IContainer container;
     IWindowManager windowManager;
     public ViewModel(IContainer container, IWindowManager windowManager)
     {
         _container = container;
         _windowManager = windowManager;
     }

     public void Dialog()
     {
          var dialog = _container.Get<ViewModel1>();
          var res = _windowManger(dialog);
          .........
     }
}
```
